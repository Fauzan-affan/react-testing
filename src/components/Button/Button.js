import React from 'react'
import styles from "./Button.module.css"

export default function Button({label}) {
    return (
        <>
            <button data-testid="button" className={styles.button}>{label}</button>
        </>
    )
}
