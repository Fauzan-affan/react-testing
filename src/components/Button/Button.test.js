import React from 'react'
import {render, cleanup} from '@testing-library/react'
import renderer from "react-test-renderer"

import Button from "./Button.js";

afterEach(cleanup)

it("renders without crashing", () => {
    render(<Button/>)
})

test("set props", () => {
    const {getByTestId} = render(<Button label="Please Click Me Guys!!!"></Button>)
    expect(getByTestId("button")).toHaveTextContent("Please Click Me Guys!!!")
})

test("set props", () => {
    const {getByTestId} = render(<Button label="Ah Elah"></Button>)
    expect(getByTestId("button")).toHaveTextContent("Ah Elah")
})

describe("check snapshot", () => {
    const tree = renderer.create(<Button label="Save Me"/>).toJSON()
    expect(tree).toMatchSnapshot()
})