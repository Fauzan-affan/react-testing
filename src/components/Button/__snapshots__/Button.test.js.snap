// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="button"
  data-testid="button"
>
  Save Me
</button>
`;
